pypesto/visualize/profiles.py

```diff
@@ -215,6 +253,12 @@ def profiles_lowlevel(
         # if we have empty profiles and more axes than profiles: skip
         if n_plots != n_fvals and fval is None:
             continue
+        # If we use colors from profiler_result.color_path,
+        # we need to take the color path of each profile
+        if isinstance(color, list) and isinstance(color[i_plot], np.ndarray):
```

Comment:
Why a `np.array` here? I no other data structure works, that should be reflected in the type hints of `profiles` and `profiles_lowlevel`, otherwise it could be a more generic sequence type

Reply:
Is the second condition indirectly checking if the colors are from `profiler_result.color_path`?

Reply:
> Is the second condition indirectly checking if the colors are from `profiler_result.color_path`?

Yeah, basically.
There are 3 cases:
- `color = None`, then `color_i` will be set to `None` for some default later in `profile_lowlevel`. This is the case if `quality_colors=False` and there's only one result and one profile_list_id (small digression here: one result can have multiple profile results, for instance, in a case when you ran profiling in all local minima you wanted. Then each of those profile results corresponds to one profile_list_id).
- `color = np.ndarray` of length 4 -- one RGBA color. This is the case if there's more than one result or more than one profile_list_id. Then the color cannot be the default one but are specified beforehand by  `process_result_list_profiles`.
- `color = list[np.ndarray[np.ndarray]]` -- this is the case if we use color_paths (quality_colors). The list iterates over profiled parameters, and each entry is the numpy array of RGBAs of length 4. 

In the first two cases, we need to pass the color directly as the color_i (color of the specific i-th profile) -- either None or just one RGBA.
In the last case, we need to select the color_path of the corresponding parameter profile we're plotting now.

And it seemed the easiest to check whether it's a list and its entry is a numpy array.
I could use `if color is not None and not isinstance(color, np.ndarray)` as well, but it's kinda the same (and I think I might've tried that first, and had a bug with a weird edge case).

Reply:
I probably should change the docstring tho... As currently it just says that it can be RGBA or None

Reply:
Suggestion:
```
color: RGBA, list[np.ndarray[RGBA]], optional
        Color for profiles in plot. In case of quality_colors=True, this is a list of
        np.ndarray[RGBA] for each profile -- one color per profile point for each profile.
```

Reply:
My issue is rather that there are two different options for coloring: the `quality_colors` and the `colors` argument of `profiles`.
1. If a user supplies a list of `np.ndarray`, that should not be interpreted as `quality_color` here
2. If the `profile_lowlevel` only takes a list of `np.ndarray`, that should be reflected in the type hint in lines 47-48 (old).

Reply:
Yes, there are now two different coloring options. But I do check and raise a ValueError if both of them are given.

`profiles_lowlevel` could previously (old code) take only a single np.ndarray (one RGBA color). It was never given a list of np.ndarray. When a user provides a list of `np.ndarray` (a list of multiple colors) then one color represents profiles of either one result or one profile_list_id -- `profiles_lowlevel` still gets only one color.

And now, the only case at which the `profiles_lowlevel` takes a list of np.ndarrays is when `quality_colors` is used.

So it should never clash. But I can put in the docstring of `colors` (47-48 old) that it cannot be used together with `quality_colors`.

I hope I haven't misunderstood you.

Reply:
After this corrections, `visualize` looks fine to me. Thanks for the contribution.